### PR TITLE
Use latest nixpkgs

### PR DESCRIPTION
**Pull request by kozross**
This relates to [this suggestion here](https://github.com/input-output-hk/iohk-nix/pull/503#issuecomment-1034952292). Ultimately, this is needed for [Schnorr signature support](https://github.com/input-output-hk/plutus/pull/4368#issuecomment-1031782137).

### CODE REVIEW

**Comment by kozross** [comment]
I'm getting a really weird build failure locally:

```
error: builder for '/nix/store/di7r7vnkakysi0ambfgpdi48qyx4qzxj-discount-2.2.7.drv' failed with exit code 2;
       last 10 log lines:
       > installing
       > install flags: SHELL=/nix/store/cfwzs61rdvvziplxh4gyq49f8gwrf42k-bash-5.1-p12/bin/bash install
       > /tmp/nix-build-discount-2.2.7.drv-0/source/config.md /nix/store/31ypra6mzlx74kcva80nn3gaxxxxcy73-discount-2.2.7/bin
       > /tmp/nix-build-discount-2.2.7.drv-0/source/config.md /nix/store/31ypra6mzlx74kcva80nn3gaxxxxcy73-discount-2.2.7/lib
       > /tmp/nix-build-discount-2.2.7.drv-0/source/config.md /nix/store/31ypra6mzlx74kcva80nn3gaxxxxcy73-discount-2.2.7/include
       > /tmp/nix-build-discount-2.2.7.drv-0/source/config.md /nix/store/31ypra6mzlx74kcva80nn3gaxxxxcy73-discount-2.2.7/lib/pkgconfig
       > /nix/store/hjymmi7cq1wy3fb1l7nyk274y5ijgnhw-coreutils-9.0/bin/install -s -m 755 markdown /nix/store/31ypra6mzlx74kcva80nn3gaxxxxcy73-discount-2.2.7/bin
       > ./librarian.sh install libmarkdown VERSION /nix/store/31ypra6mzlx74kcva80nn3gaxxxxcy73-discount-2.2.7/lib
       > /sbin/ldconfig: Can't create temporary cache file /etc/ld.so.cache~: Permission denied
       > make: *** [Makefile:43: install] Error 1
    
   For full logs, run 'nix log /nix/store/di7r7vnkakysi0ambfgpdi48qyx4qzxj-discount-2.2.7.drv'.
```

This clearly seems to be an issue locally, but I have no idea why this is happening or how to diagnose it.

**Comment by michaelpj** [comment]
Hydra has some odd eval errors. To minimize the change surface, I suggest using the same rev as in https://github.com/input-output-hk/haskell.nix/pull/1371

**Comment by kozross** [comment]
@michaelpj I tried (what I understood was) your suggestion, but I seem to hit the same issue. I've pushed that as a commit to this branch: please let me know if I have misunderstood something.

**Comment by michaelpj** [comment]
It was just a hope :sob:

File: nix/sources.json [hunk 1]
```diff
@@ -44,7 +44,7 @@
       "homepage": null,
       "owner": "NixOS",
       "repo": "nixpkgs",
-      "rev": "d68537fe60c16405c6673f6a06ca70607e443fb9",
+      "rev": "1882c6b7368fd284ad01b0a5b5601ef136321292",
```
Comment (michaelpj) [inline nix/sources.json:47]: This doesn't change the `sha256` so it does nothing. I guess you just edited this by hand instead of using `niv update` :)

Comment (michaelpj) [reply nix/sources.json:47]: I fixed this locally and things seemed to get a lot further before I got bored and killed it.

**Comment by michaelpj** [comment]
I'm not sure what you did this time. It looks like you updated `haskell.nix` but didn't add `nixpkgs` at the revision that you wanted?

**Comment by michaelpj** [comment]
I took a stab at it here, let's see how it goes: https://github.com/input-output-hk/cardano-base/pull/257

**Pull request closed without merging**